projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/index.ts

```diff
@@ -1,24 +1,22 @@
-import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
+import useRedBubbleQuery from '../../data/use-red-bubble-query';
 import useBackupNeedsAttentionNotice from './use-backup-needs-attention-notice';
 import useBadInstallNotice from './use-bad-install-notice';
 import useConnectionErrorsNotice from './use-connection-errors-notice';
-import useDeprecateFeatureNotice from './use-deprecate-feature-notice';
 import useExpiringPlansNotice from './use-expiring-plans-notice';
 import usePaidPlanNeedsPluginInstallActivationNotice from './use-paid-plan-needs-plugin-install-activation-notice';
 import useProtectThreatsDetectedNotice from './use-protect-threats-detected-notice';
 import useSiteConnectionNotice from './use-site-connection-notice';
 
 const useNotificationWatcher = () => {
-	const { redBubbleAlerts } = getMyJetpackWindowInitialState();
+	const { isLoading, data: redBubbleAlerts } = useRedBubbleQuery();
 
-	usePaidPlanNeedsPluginInstallActivationNotice( redBubbleAlerts );
-	useProtectThreatsDetectedNotice( redBubbleAlerts );
-	useExpiringPlansNotice( redBubbleAlerts );
-	useBackupNeedsAttentionNotice( redBubbleAlerts );
-	useDeprecateFeatureNotice( redBubbleAlerts );
+	usePaidPlanNeedsPluginInstallActivationNotice( redBubbleAlerts, isLoading );
+	useProtectThreatsDetectedNotice( redBubbleAlerts, isLoading );
+	useExpiringPlansNotice( redBubbleAlerts, isLoading );
+	useBackupNeedsAttentionNotice( redBubbleAlerts, isLoading );
 	useConnectionErrorsNotice();
-	useSiteConnectionNotice( redBubbleAlerts );
-	useBadInstallNotice( redBubbleAlerts );
+	useSiteConnectionNotice( redBubbleAlerts, isLoading );
+	useBadInstallNotice( redBubbleAlerts, isLoading );
 };
```

Comment:
It feels like we are again passing props to many places. Do you think it would be useful to introduce Context on red bubble query?

Reply:
In this specific case, I don't think that it's bad to be passing one more prop, especially since all the hooks will have the same props. I think we could make an argument to create contexts for several points of data. Product data could be in context, purchases, etc. And I think that may be the direction we head in the future. However I'd argue against doing it here to avoid making this PR larger than it already is.

I also like to avoid passing props so I understand your concern here, but I think in this specific case it seems reasonable 🤔